File: rmt/src/test/scala/de/flower/test/AbstractWicketTests.scala
package de.flower.test

import org.apache.wicket.protocol.http.WebApplication
import org.springframework.context.ApplicationContext
import org.testng.annotations.AfterMethod
import org.testng.annotations.BeforeMethod
import org.mockito.Mockito._
import scala.collection.JavaConversions._
import org.apache.wicket.Component
import org.apache.wicket.util.tester.{FormTester, WicketTester, WicketTesterHelper}

/**
 * Base class for testing wicket components using TestNG.
 */
abstract class AbstractWicketTests extends AbstractIntegrationTests {

    @BeforeMethod def init: Unit = {
        createTester(applicationContext)
    }

    @AfterMethod def cleanup: Unit = {
        validateMockitoUsage
    }

    protected def createWebApp(ctx: ApplicationContext, flag: Boolean): WebApplication

    private def createTester(ctx: ApplicationContext): Unit = {
        val webApp: WebApplication = createWebApp(ctx, true)
        wicketTester = new RMTWicketTester(webApp)
    }



    protected var wicketTester: RMTWicketTester = null

}

File: rmt/src/test/java/de/flower/rmt/test/AbstractWicketTests.java
package de.flower.rmt.test;

import de.flower.rmt.ui.app.TestApplication;
import org.apache.wicket.protocol.http.WebApplication;
import org.mockito.Mockito;
import org.springframework.context.ApplicationContext;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;

/**
 * @author flowerrrr
 */

public abstract class AbstractWicketTests extends AbstractIntegrationTests {

    protected RMTWicketTester wicketTester = null;

    protected WebApplication createWebApp(ApplicationContext ctx) {
        return new TestApplication(ctx);
    }

    @BeforeMethod
    public void init() {
        createTester(applicationContext);
    }

    @AfterMethod
    public void cleanup() {
        Mockito.validateMockitoUsage();
    }

    private void createTester(ApplicationContext ctx) {
        WebApplication webApp = createWebApp(ctx);
        wicketTester = new RMTWicketTester(webApp);
    }

    public abstract void testRender();
}